build/gen_stub.php

```diff
@@ -342,11 +350,25 @@ class FuncInfo {
     public $cond;
 
     public function __construct(
-        string $name, ?string $className, ?string $alias, bool $isDeprecated, array $args, ReturnInfo $return,
-        int $numRequiredArgs, ?string $cond
+        string $name,
+        ?string $className,
+        ?string $visibility,
+        bool $isAbstract,
+        bool $isStatic,
+        bool $isFinal,
```

Comment:
Rather than adding four separate members, I'd suggest passing through `$method->flags` instead. It includes visibility, abstract, static and final flags.

Reply:
The main reason why I chose to add them separately is that we need to have an `$isAbstract` property to evaluate if `ZEND_ME` or `ZEND_ABSTRACT_ME` should be used. Also, I like the fact that the `FuncInfo` encapsulates the logic of generating the flags value (possibly by extending the method modifier flags with the deprecated flag), just like it does for the arg info. Another approach I was considering is to put the method modifier flags to a separate class, but I finally went with the lazy way.

I'm ok with either solutions, so please choose one. My preference though would be to stay with the current one or use composition.

Reply:
To be clear, I meant passing the integer flags value (not the string representation), which is a bitset of `Class_::MODIFIER_*`: https://github.com/nikic/PHP-Parser/blob/bd43ec7152eaaab3bd8c6d0aa95ceeb1df8ee120/lib/PhpParser/Node/Stmt/Class_.php#L10-L15 You would check for abstract using `$this->flags & Class_::MODIFIER_ABSTRACT`.

Reply:
Oops, I see now.